app/models/channel.rb

```diff
@@ -267,6 +267,10 @@ def uphold_connection
     @uphold_connection ||= uphold_connection_for_channel.detect { |connection| connection.currency == publisher.uphold_connection.default_currency }
   end
 
+  def register_channel_for_promo
+    Promo::RegisterChannelForPromoJob.perform_now(channel_id: id)
```

Comment:
I believe moving this simply pass the `id` is going to lead to a race condition due to the `after_save` callback on this method.

Before we were passing `self` into the registration job which wasn't reliant on the database. 

Now that we're passing the `id` the `Channel.new.save` transaction may not yet be committed when we call `perform_now` and so when we call `Channel.find` in the `RegisterChannelForPromoJob` the lookup will fail. 

Since this is done synchronously the job never enters the queue and so this will actually fail to let people create new channels. 

So I think the best approach is to move this to an `after_commit` callback